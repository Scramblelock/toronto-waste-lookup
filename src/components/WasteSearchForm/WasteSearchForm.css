.form {
  width: 100%;
  display: flex;
  flex-flow: row nowrap;
  margin-bottom: 20px;
}

.form[type=submit] {
  -webkit-appearance: none;
}

.form input {
  border-radius: 4px;
  margin-left: 1em;
  margin-right: 1em;
}

.form input[type=text] {
  border: 1px solid #000;
  padding: .75rem;
  font-size: 25px;
  flex: 1;
}

.form button {
	padding: 0;
	margin-right: 1.5em;
	cursor: pointer;
	border-radius: 5%;
	background-color: #22955f;
}

.form button:hover {
  background-color: #47c496;
}