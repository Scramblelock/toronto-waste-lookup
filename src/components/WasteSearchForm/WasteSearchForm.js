import React from 'react';
import './WasteSearchForm.css';
import magnifyingGlass from './icons8-search-60.png';

const WasteSearchForm = () => {
	return (
		<form className='form'>
      <input 
      	type='text'
      	placeholder='Enter a keyword'
      	required
      />
      <button className='grow shadow-5' type='submit'>
      	<img src={magnifyingGlass} alt='magnifyingGlass'/>
      </button>
    </form>
  );  
}

export default WasteSearchForm;
