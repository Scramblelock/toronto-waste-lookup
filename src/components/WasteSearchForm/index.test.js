import { shallow } from 'enzyme';
import React from 'react';
import WasteSearchForm from './index';

it('expect to render WasteSearchForm component', () => {
	expect(shallow(<WasteSearchForm />).length).toEqual(1);
})

it('expect it to render WasteSearchForm component', () => {
	expect(shallow(<WasteSearchForm />)).toMatchSnapshot();
})
