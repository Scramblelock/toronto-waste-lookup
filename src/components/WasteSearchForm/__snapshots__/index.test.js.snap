// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect it to render WasteSearchForm component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <WasteSearchForm />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <label
          className="visuallyhidden"
          htmlFor="search"
        >
          Search:
        </label>,
        <input
          placeholder="Enter a keyword"
          required={true}
          type="text"
        />,
        <button
          aria-label="searchbutton"
          className="grow shadow-5"
        >
          <img
            alt="magnifyingGlass"
            src="icons8-search-60.png"
          />
        </button>,
      ],
      "className": "form",
      "onSubmit": undefined,
      "type": "submit",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Search:",
          "className": "visuallyhidden",
          "htmlFor": "search",
        },
        "ref": null,
        "rendered": "Search:",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": undefined,
          "placeholder": "Enter a keyword",
          "required": true,
          "type": "text",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "aria-label": "searchbutton",
          "children": <img
            alt="magnifyingGlass"
            src="icons8-search-60.png"
          />,
          "className": "grow shadow-5",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "alt": "magnifyingGlass",
            "src": "icons8-search-60.png",
          },
          "ref": null,
          "rendered": null,
          "type": "img",
        },
        "type": "button",
      },
    ],
    "type": "form",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <label
            className="visuallyhidden"
            htmlFor="search"
          >
            Search:
          </label>,
          <input
            placeholder="Enter a keyword"
            required={true}
            type="text"
          />,
          <button
            aria-label="searchbutton"
            className="grow shadow-5"
          >
            <img
              alt="magnifyingGlass"
              src="icons8-search-60.png"
            />
          </button>,
        ],
        "className": "form",
        "onSubmit": undefined,
        "type": "submit",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Search:",
            "className": "visuallyhidden",
            "htmlFor": "search",
          },
          "ref": null,
          "rendered": "Search:",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "onChange": undefined,
            "placeholder": "Enter a keyword",
            "required": true,
            "type": "text",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "aria-label": "searchbutton",
            "children": <img
              alt="magnifyingGlass"
              src="icons8-search-60.png"
            />,
            "className": "grow shadow-5",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "magnifyingGlass",
              "src": "icons8-search-60.png",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "button",
        },
      ],
      "type": "form",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
